app/controllers/nonsipp/loansmadeoroutstanding/LoansCYAController.scala

```diff
@@ -163,11 +164,23 @@ class LoansCYAController @Inject()(
 
   def onSubmit(srn: Srn, index: Max5000, mode: Mode): Action[AnyContent] =
     identifyAndRequireData(srn).async { implicit request =>
+      val userAnswersWithJourneysCompleted = buildCompletedLoanIndexes(srn, index).map(
+        _.foldLeft(Try(request.userAnswers))(
+          (userAnswersTry, loanIndex) => userAnswersTry.flatMap(_.set(LoanCompleted(srn, loanIndex), SectionCompleted))
+        )
+      )
+
       for {
+        _ <- userAnswersWithJourneysCompleted match {
+          case Left(result) => Future.successful(result)
+          case Right(_) => Future.unit
+        }
+
         updatedAnswers <- Future.fromTry(
           request.userAnswers
             .set(LoansMadeOrOutstandingPage(srn), true)
             .set(LoanCompleted(srn, index), SectionCompleted)
+            .set(LoansProgress(srn, index), SectionJourneyStatus.Completed)
```

Comment:
This should happen on the last page in the regular journey (before CYA page) and on the last page of the pre-population journey (only if it is a pre-population).